rust/src/graph/higher_order_queries.rs

```diff
@@ -84,7 +84,7 @@ impl Graph {
             )
     }
 
-    fn generate_module_permutations(
+    fn generate_illegal_import_permutations_for_layers(
```

Comment:
Definitely better - might be worth adding a comment to explain what the tuple corresponds to (e.g. is the first one the source and the second one the module that should not be imported by it)?